vital/types/detected_object_type.h

```diff
@@ -74,6 +74,9 @@ class VITAL_EXPORT detected_object_type
 public:
   static const double INVALID_SCORE;
 
+  using class_map_t = std::map< std::string const*, double >;
```

Comment:
> he just changed it to a typedef (using) so it was clearer.

...or, so I could write the iterator type without it being obscenely long :wink:. But yes, the implementation uses pointers to strings (the instances are stored in the master types table, I believe), so unfortunately this implementation detail is exposed. See my previous comment; if this is an issue, I believe an iterator adapter can be created to hide it...